google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -143,18 +156,45 @@ public GapicSpannerRpc(SpannerOptions options) throws IOException {
   @Override
   public Paginated<InstanceConfig> listInstanceConfigs(int pageSize, @Nullable String pageToken)
       throws SpannerException {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    ListInstanceConfigsRequest.Builder requestBuilder = 
+        ListInstanceConfigsRequest.newBuilder().setParent(projectName).setPageSize(pageSize);
+    if (pageToken != null) {
+      requestBuilder.setPageToken(pageToken);
+    }
+    ListInstanceConfigsRequest request = requestBuilder.build();
+
+    // TODO: put projectName in metadata
+    ListInstanceConfigsResponse response = 
+        get(instanceStub.listInstanceConfigsCallable().futureCall(request));
+    return new Paginated<>(response.getInstanceConfigsList(), response.getNextPageToken());
   }
 
   @Override
   public InstanceConfig getInstanceConfig(String instanceConfigName) throws SpannerException {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    GetInstanceConfigRequest request = 
+        GetInstanceConfigRequest.newBuilder().setName(instanceConfigName).build();
+
+    // TODO: put projectName in metadata
+    return get(instanceStub.getInstanceConfigCallable().futureCall(request));
   }
 
   @Override
   public Paginated<Instance> listInstances(
       int pageSize, @Nullable String pageToken, @Nullable String filter) throws SpannerException {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    ListInstancesRequest.Builder requestBuilder =
+        ListInstancesRequest.newBuilder().setParent(projectName).setPageSize(pageSize);
+    if (pageToken != null) {
+      requestBuilder.setPageToken(pageToken);
+    }
```

Comment:
Why do we check for null for some Strings but not for others (e.g. `instanceName` in the `getInstance` function below)? If the proto-generated class is tolerate to nulls, then we shouldn't check anywhere.

Reply:
I did it the same way `GrpcSpannerRpc` implemented it but I am not super confident in it. Not sure if I understand correctly, both `ListInstancesRequest.Builder` and `GetInstanceRequest.Builder` do not allow null values (would throw a `NullPointerException`), but `pageToken` and `filter` are actually allowed to be null by API doc (the doc is for REST though): https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances/list

Reply:
Offline resolution: the pageToken/filter parameters are optional, so we should allow nulls.

Reply:
I could be wrong, but I don't think protobuf lets you set anything to null. IIRC, setting to null will throw `NullPointerException`.